src/ncp/ncp_base_ftd.cpp

```diff
@@ -1438,6 +1438,34 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_INFRA_IF_STATE>(void)
     return error;
 }
 
+template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_INFRA_IF_RECV_ICMP6>(void)
+{
+    otError                 error = OT_ERROR_NONE;
+    spinel_ipv6_icmp_type_t msgType;
+    uint32_t                infraIfIndex;
+    const otIp6Address     *address;
+    const uint8_t          *icmp6Data = nullptr;
+    uint16_t                len;
+
+    SuccessOrExit(error = mDecoder.ReadUint32(infraIfIndex));
+    VerifyOrExit(mInfraIfIndex == infraIfIndex, error = OT_ERROR_DROP);
+    SuccessOrExit(error = mDecoder.ReadUint8(msgType));
+    SuccessOrExit(error = mDecoder.ReadIp6Address(address));
+    SuccessOrExit(error = mDecoder.ReadData(icmp6Data, len));
+
+    if (msgType == SPINEL_IPV6_ICMP_TYPE_ND)
```

Comment:
this is probably not necessary, as `otPlatInfraIfRecvIcmp6Nd` will check the message type.

Reply:
`otPlatInfraIfRecvIcmp6Nd()` needs to check whether the packet is ND anyway. If you check the current code, it doesn't verify whether it's ND before delivering them to this API either. We just trust the kernel will do the right thing.

Reply:
Make sense. We can update `otPlatInfraIfRecvIcmp6Nd` to `otPlatInfraIfRecvIcmp6` when there are other icmp6 messages. I added a comment here to avoid confusion.

Reply:
`otPlatInfraIfRecvIcmp6Nd()` explicitly states that the platform should call it with ND messages:

```c++
/**
 * The infra interface driver calls this method to notify OpenThread
 * that an ICMPv6 Neighbor Discovery message is received.   
```

If a platform implementation doesn't adhere to this, I recommend updating or fixing the platform code.

> We can update otPlatInfraIfRecvIcmp6Nd to otPlatInfraIfRecvIcmp6

We cannot and should not rename or change existing APIs to ensure backward compatibility. OT public/platform APIs are our contract with vendors/users. We need a very strong justification for any changes and only allow changes after exploring all other options/solutions.

Reply:
Make sense. I'll keep that in mind. Then I think for this PR, we just use the original approach: using `SPINEL_PROP_INFRA_IF_RECV_ICMP6_ND`. Considering there are no foreseeable case for non-ND messages, we can add another property in the future if there is any.